examples/multisig_plain/lib.rs

```diff
@@ -0,0 +1,231 @@
+            self.ensure_from_wallet();
+            self.take_transaction(id);
+        }
+
+        #[ink(message)]
+        fn confirm_transaction(&mut self, id: TransactionId) {
+            self.ensure_from_owner();
+            self.ensure_transaction_exists(id);
+            self.internal_confirm(id);
+        }
+
+        #[ink(message)]
+        fn revoke_confirmation(&mut self, id: TransactionId) {
+            self.ensure_from_owner();
+            if self
+                .confirmations
+                .remove(&(id, self.env().caller()))
+                .is_some()
+            {
+                self.confirmation_count
+                    .mutate_with(&id, |count| *count -= 1);
+            }
+        }
+
+        #[ink(message)]
+        fn execute_transaction(&mut self, id: TransactionId) -> Result<(), ()> {
+            self.ensure_confirmed(id);
+            let t = self.take_transaction(id).unwrap();
+            env::call::CallParams::<MyEnv, ()>::invoke(t.callee, t.selector.into())
+                .gas_limit(t.gas_limit)
+                .transferred_value(t.transferred_value)
+                .push_arg(&CallInput(&t.input))
+                .fire()
+                .map(|_| ())
+                .map_err(|_| ())
+        }
+
+        fn internal_confirm(&mut self, id: TransactionId) {
+            if self
+                .confirmations
+                .insert((id, self.env().caller()), ())
+                .is_none()
+            {
+                self.confirmation_count
+                    .mutate_with(&id, |count| *count += 1);
+            }
+        }
+
+        fn owner_index(&self, owner: &AccountId) -> u32 {
+            self.owners.iter().position(|x| *x == *owner).unwrap() as u32
+        }
+
+        fn take_transaction(&mut self, id: TransactionId) -> Option<Transaction> {
+            let transaction = self.transactions.take(id);
+            if transaction.is_some() {
+                self.clean_transaction_confirmations(id);
+            }
+            transaction
+        }
+
+        fn clean_owner_confirmations(&mut self, owner: &AccountId) {
+            for (id, _) in self.transactions.iter() {
+                if self.confirmations.remove(&(id, *owner)).is_some() {
+                    self.confirmation_count
+                        .mutate_with(&id, |count| *count -= 1);
+                }
+            }
+        }
+
+        fn clean_transaction_confirmations(&mut self, transaction: TransactionId) {
+            for owner in self.owners.iter() {
+                self.confirmations.remove(&(transaction, *owner));
+            }
+            self.confirmation_count.remove(&transaction);
+        }
+
+        fn ensure_confirmed(&self, id: TransactionId) {
+            assert!(self.confirmation_count.get(&id).unwrap() >= self.required.get());
+        }
+
+        fn ensure_transaction_exists(&self, id: TransactionId) {
+            self.transactions.get(id).unwrap();
```

Comment:
Are you sure about that? Basically what we are assuming here is: "Okay the user gave us an invalid id. We do not care: Just rollback". So there is no real proof behind that that this cannot happen. It is totally plausible that this happens. It's not like in runtime where we actually `expect` it to not happen.

Reply:
Well that's what you would write into `expect` to make clear that panicking upon this situation is actually the way this API works. Otherwise contract devs looking at this code might think that this is a logic error etc.

Reply:
That seems reasonable. I will remove all unwraps.

Reply:
Done. Reopen if necessary.